### DISCUSSION TITLE
Access denied for user 'ejbca'@'localhost' (using password: YES)

### DISCUSSION THREAD

**Discussion by husmannm** [Keyfactor/ejbca-ce#189] [Q&A] [answered]
Hi
During Appserver Startup I get the error message "....Access denied for user 'ejbca'@'localhost' (using password: YES)..." (from log "/opt/wildfly/standalone/log/server.log")

If I try to login to the database directly (from CLI using "mysql -u ejbca -p"), I can login as user ejbca@localhost.

MariaDB [ejbca]> select current_user();
+-----------------+
| current_user()  |
+-----------------+
| ejbca@localhost |
+-----------------+
1 row in set (0.000 sec)

Username and password are configured in database.properties and deployed with 'ant deployear'

My questions:
Where are the current settings for the connections visible (used by the App)?
How can I test the connection?

Info:
Server: Debian Bullseye 11.5
DB: MariaDB 10.5.18
Wildfly: 26.0.0
EJBCA: CE 7.10.0.2

**Comment by primetomas** [comment, marked as answer]
You don't need to set any database url or password in database.properties, only the database type.
The database url and password is set when you configure the application server, when adding the datasource.
https://doc.primekey.com/ejbca/ejbca-installation/application-servers/wildfly-26#WildFly26-AddaDatasource

Depending on which instruction you use. For simpler configurations the password was in standalone.xml in wildfly, but for the later instructions a WildFly credential store is used, so the password has to be changed in that. Or the datasource reconfigured with a new password in the credential store.

**Reply by husmannm** [reply]
Thank you, you pointed me to the right direction!
